src/main/xsl/preprocess/topicpullImpl.xsl

```diff
@@ -100,31 +187,36 @@ mode="topicpull:figure-linktext" and mode="topicpull:table-linktext"
       <!--@type|@format|@scope|@importance|@role-->
 
       <!--need to create type, format, scope variables regardless of whether they exist, for passing as a parameter to getstuff template-->
-      <xsl:variable name="type" as="xs:string">
-        <xsl:call-template name="topicpull:inherit"><xsl:with-param name="attrib">type</xsl:with-param></xsl:call-template>
-      </xsl:variable>
-      <xsl:variable name="format" as="xs:string">
-        <xsl:call-template name="topicpull:inherit"><xsl:with-param name="attrib">format</xsl:with-param></xsl:call-template>
-      </xsl:variable>
-      <xsl:variable name="scope" as="xs:string">
-        <xsl:call-template name="topicpull:inherit"><xsl:with-param name="attrib">scope</xsl:with-param></xsl:call-template>
-      </xsl:variable>
-
-      <xsl:if test="not(@type) and $type!='#none#'">
+      <xsl:variable name="type" as="xs:string?" select="dita-ot:get-link-target-type(.)"/>
+      
+      <xsl:variable name="format" as="xs:string?" select="dita-ot:get-link-format(.)"/>
+      <xsl:variable name="scope" as="xs:string?" select="dita-ot:get-link-scope(.)"/>
+      
+      <xsl:if test="not(@type) and $type">
         <xsl:attribute name="type"><xsl:value-of select="$type"/></xsl:attribute>
       </xsl:if>
-      <xsl:if test="not(@format) and $format!='#none#'">
+      <xsl:if test="not(@format) and $format">
```

Comment:
IMO we should use `exists` and `empty` to make code a little more explicit. So this could be `empty(@format) and exists($format)`, but I don't feel very strongly about it being in this PR.